demo/demo.h

```diff
@@ -37,37 +44,207 @@ DECLARE_int32(demo_port);
+    HTTP(port).Register("/layout/meta", [](Request r) {
+      r.connection.SendHTTPResponse(ExampleMeta(),
+                                    "meta",
+                                    HTTPResponseCode::OK,
+                                    "application/json; charset=utf-8",
+                                    {{"Connection", "close"}, {"Access-Control-Allow-Origin", "*"}});
+    });
+    HTTP(port).Register("/layout", [](Request r) {
+      LayoutItem layout;
+      LayoutItem row;
+      layout.col.push_back(row);
+      r.connection.SendHTTPResponse(layout,
+                                    "layout",
+                                    HTTPResponseCode::OK,
+                                    "application/json; charset=utf-8",
+                                    {{"Connection", "close"}, {"Access-Control-Allow-Origin", "*"}});
+    });
+    // The "./static/" directory should be a symlink to "Web/build" or "Web/build-dev".
+    FileSystem::ScanDir("./static/", [&port](const std::string& filename) {
+      std::string filepath = "./static/" + filename;
+      std::string fileurl = "/static/" + filename;
+      // The default route points to the frontend entry point file.
+      if (filename == "index.html") {
+        fileurl.assign("/");
+      }
       // Hack with string ownership.
       // TODO(dkorolev): Make this cleaner.
-      std::string* ps = new std::string();
-      *ps = FileSystem::ReadFileAsString("static/" + fn);
-      HTTP(port).Register("/static/" + fn, [ps](Request r) {
-        r.connection.SendHTTPResponse(*ps, HTTPResponseCode::OK, "text/plain");
+      std::string* contentType = new std::string(DemoServer::GetFileMimeType(filename));
+      std::string* content = new std::string();
+      *content = FileSystem::ReadFileAsString(filepath);
+      HTTP(port).Register(fileurl, [content, contentType](Request r) {
+        r.connection.SendHTTPResponse(*content, HTTPResponseCode::OK, *contentType);
       });
     });
   }
 
   void Join() {
-    using bricks::net::api::HTTP;
-
     printf("Listening on port %d\n", port_);
     HTTP(port_).Join();
     printf("Done.\n");
   }
 
+ private:
   State state_;
   const int port_;
+
+  /**
+   * Gets the file extension from the file name or path.
+   */
+  static const std::string GetFileExtension(const std::string& filename) {
+    size_t fileextIndex = filename.find_last_of("/\\.");
+
+    if (fileextIndex == std::string::npos || filename[fileextIndex] != '.') {
+      return "";
+    }
+
+    // Add 1 to skip the dot.
+    std::string fileext = filename.substr(fileextIndex + 1);
+
+    return fileext;
+  }
+
+  /**
+   * Detects MIME type by the file extension.
+   */
+  static const std::string GetFileMimeType(const std::string& filename) {
+    std::string fileext = DemoServer::GetFileExtension(filename);
+
+    std::transform(fileext.begin(), fileext.end(), fileext.begin(), ::tolower);
+
+    std::map<std::string, std::string> mimeTypesByExtension = {
```

Comment:
```const``` for sure, and ```static const``` to only initialize it once per binary run, not on every call to ```GetFileMimeType()```.